src/app/pages/common/ix-forms/services/ix-formatter.service.ts

```diff
@@ -0,0 +1,183 @@
+    } else if (value.replace(/\s/g, '').match(/[^0-9]/g) === null) {
+      formatted = IxFormatter.convertBytestoHumanReadable(value.replace(/\s/g, ''), 0);
+    } else {
+      formatted = IxFormatter.humanReadable;
+    }
+    return formatted;
+  };
+
+  static readonly memorySizeParsing: (val: string) => number = (value: string) => {
+    value = value.toString();
+    if (!value) {
+      return NaN;
+    }
+    let humanStringToNum = IxFormatter.convertHumanStringToNum(value);
+    if (humanStringToNum === Number(value)) {
+      value += 'mb';
+      humanStringToNum = IxFormatter.convertHumanStringToNum(value);
+    }
+    return humanStringToNum;
+  };
+
+  // Converts a number from bytes to the most natural human readable format
+  static readonly convertBytestoHumanReadable = (
+    rawBytes: number | string,
+    decimalPlaces?: number,
+    minUnits?: string,
+    hideBytes?: boolean,
+  ): string => {
+    let i = 0;
+    let units;
+    let bytes = Number(rawBytes);
+
+    const dec = decimalPlaces !== undefined ? decimalPlaces : 2;
+    if (bytes >= 1024) {
+      do {
+        bytes = bytes / 1024;
+        i++;
+      } while (bytes >= 1024 && i < 4);
+      units = IxFormatter.IECUnits[i];
+    } else if (minUnits) {
+      units = minUnits;
+    } else {
+      units = hideBytes ? '' : 'B';
+    }
+    return `${bytes.toFixed(dec)} ${units}`;
+  };
+
+  // sample data, input and return values
+  // input       normalized       number value
+  // '12345'     '12345'          12345
+  // '512x'      ''               NaN
+  // '0'         '0'              0
+  // '0b'        ''               NaN
+  // '',         '0'              0
+  // '4MB',      '4 MiB'          4*1024**2 (4,194,304)
+  // '16KiB'     '16 KiB'         16*1024   (16,384)
+  // 'g'         ''               NaN
+  // ' t1'       ''               NaN
+  // '   5   m'  '5 MiB'          5*1024**2 (5,242,880)
+  // '1m',       '1 MiB'          1024**2   (1,048,576)
+  // '    T'     ''               NaN
+  // '2 MiB  '   '2 MiB'          2*1024**2 (2,097,152)
+  // '2 MiB x8'  ''               NaN
+  // '256 k'     '256 KiB'        256*1024  (262,144)
+  // 'm4m k'     ''               NaN
+  // '4m k'      ''               NaN
+  // '1.2m'      ''               NaN
+  // '12k4'      ''               NaN
+  // '12.4k'     ''               NaN
+  // ' 10G'      '10 GiB'         10*1024**3 (10,737,418,240)
+
+  // hstr = the human string from the form;
+  // dec = allow decimals;
+  // allowedUnits (optional) should include any or all of 'kmgtp', the first letters of KiB, Mib, etc. The first letter
+  // is used as the default, so for 'gtp', an entered value of 256 becomes 256 GiB. If you don't pass in allowedUnits,
+  // all of the above are accepted AND no unit is attached to an unlabeled number, so 256 is considered 256 bytes.
+  static readonly convertHumanStringToNum = (hstr: string, dec = false, allowedUnits?: string): number => {
```

Comment:
I don't know if these functions that you copied are meant to be ready for review or not.

If they are ready for review, I'd make the following changes:
1. Make them class methods as opposed to static.
2. Make them pure and not store anything in `humanReadable` property.
2. Return `null` if parsing failed.
3. Add a couple of unit tests for them.
4. Checked typing for `allowedUnits`.
5. Reformatted JSDoc